packages/dev/core/src/Meshes/mesh.ts

```diff
@@ -2202,6 +2202,43 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
         this._instanceDataStorage.previousBatch = null;
     }
 
+    /**
+     * Triggers the draw call for the mesh (or a submesh), for a specific render pass id
+     * @param renderPassId defines the render pass id to use to draw the mesh / submesh. If not provided, use the current renderPassId of the engine.
+     * @param enableAlphaMode defines if alpha mode can be changed (default: false)
+     * @param effectiveMeshReplacement defines an optional mesh used to provide info for the rendering (default: undefined)
+     * @param subMesh defines the subMesh to render. If not provided, draw all mesh submeshes (default: undefined)
+     * @param checkFrustumCulling defines if frustum culling must be checked (default: true). If you know the mesh is in the frustum (or if you don't care!), you can pass false to optimize.
+     * @returns the current mesh
+     */
+    public renderWithRenderPassId(renderPassId?: number, enableAlphaMode?: boolean, effectiveMeshReplacement?: AbstractMesh, subMesh?: SubMesh, checkFrustumCulling = true) {
+        const engine = this._scene.getEngine();
+        const currentRenderPassId = engine.currentRenderPassId;
+
+        if (renderPassId !== undefined) {
+            engine.currentRenderPassId = renderPassId;
+        }
+
+        if (subMesh) {
+            if (!checkFrustumCulling || (checkFrustumCulling && subMesh.isInFrustum(this._scene._frustumPlanes))) {
+                this.render(subMesh, !!enableAlphaMode, effectiveMeshReplacement);
+            }
+        } else {
+            for (let s = 0; s < this.subMeshes.length; s++) {
+                const subMesh = this.subMeshes[s];
+                if (!checkFrustumCulling || (checkFrustumCulling && subMesh.isInFrustum(this._scene._frustumPlanes))) {
+                    this.render(subMesh, !!enableAlphaMode, effectiveMeshReplacement);
+                }
+            }
+        }
+
+        if (renderPassId !== undefined) {
+            engine.currentRenderPassId = currentRenderPassId;
+        }
+
+        return this;

```

Comment:
why are you returning the mesh? is there a need for chaining?

Reply:
No, but I modeled this after the `render` method, that's why I used the same parameters than this function and returned the same thing (but I agree it is not needed).